boofuzz/primitives/fuzz_list.py

```diff
@@ -0,0 +1,125 @@
+import random
+import glob
+
+from .base_primitive import BasePrimitive
+
+
+class FuzzList(BasePrimitive):
+    # store fuzz_library as a class variable to avoid copying the ~70MB structure across each instantiated primitive.
+    _fuzz_library = []
+
+    def __init__(self, value, size=-1, padding="\x00", encoding="ascii", fuzzable=True, max_len=0, name=None, filename=None):
+        """
+        Primitive that cycles through a list of "bad" values from a file. The primitive take filename and open the file to read
+        the values to use in fuzzing process.
+
+        @type  value:    str
+        @param value:    Default string value
+        @type  size:     int
+        @param size:     (Optional, def=-1) Static size of this field, leave -1 for dynamic.
+        @type  padding:  chr
+        @param padding:  (Optional, def="\\x00") Value to use as padding to fill static field size.
+        @type  encoding: str
+        @param encoding: (Optional, def="ascii") String encoding, ex: utf_16_le for Microsoft Unicode.
+        @type  fuzzable: bool
+        @param fuzzable: (Optional, def=True) Enable/disable fuzzing of this primitive
+        @type  max_len:  int
+        @param max_len:  (Optional, def=0) Maximum string length
+        @type  name:     str
+        @param name:     (Optional, def=None) Specifying a name gives you direct access to a primitive
+        @type  filename: str
+        @param filename: Filename pattern to load all fuzz value
+        """
+
+        super(FuzzList, self).__init__()
+
+        self._value = self._original_value = value
+        self.size = size
+        self.padding = padding
+        self.encoding = encoding
+        self._fuzzable = fuzzable
+        self._name = name
+        self._filename = filename
+        self.this_library = []
+        list_of_files = glob.glob(self._filename)
+        for fname in list_of_files:
+            _file_handle = open(fname,"r")
+            self.this_library.append(_file_handle.readlines())
+            _file_handle.close()
+
+        # TODO: Make this more clear
+        if max_len > 0:
+            # If any of our strings are over max_len
+            if any(len(s) > max_len for s in self.this_library):
+                # Pull out only the ones that aren't
+                self.this_library = list(set([s[:max_len] for s in self.this_library]))
```

Comment:
Lines 50-54 can be deleted. Line 55 by itself will be enough, since it is basically a filter. If there is nothing larger than `max_len`, the statement will simply have no impact.